lib/pipekit/field_repository.rb

```diff
@@ -0,0 +1,25 @@
+module Pipekit
+  module FieldRepository
+      def get_by_key(key)
+        key = Config.field(parent_resource, key)
+        search_fields("key", key)
+      end
+
+      def get_by_name(name)
+        search_fields("name", name)
+      end
+
+      private
+
+      def search_fields(field_element, value)
+        result = request.get.select { |element| element[field_element] == value }
```

Comment:
Urgh good question will have to see what's out there in the wild, although I suspect it does actually happen, I'd be interested to see how something like MongoId does it as they're normally a decent code base